src/plugins/rovpilot/index.js

```diff
@@ -1,63 +1,246 @@
-function rovpilot(name, deps) {
-  console.log('This is where rovpilot code would execute in the node process.');
-  deps.io.sockets.on('connection', function (socket) {
-    socket.on('escs_poweron', function () {
-      deps.rov.send('escp(1)');
-      console.log('escp(1) sent');
-    });
-    socket.on('escs_poweroff', function () {
-      deps.rov.send('escp(0)');
-      console.log('escp(0) sent');
-    });
-    socket.on('holdHeading_toggle', function () {
-      deps.rov.send('holdHeading_toggle()');
-      console.log('holdHeading_toggle() sent');
+(function() {
+  var DISABLED = 'DISABLED';
+
+  var ROVPilot = function ROVPilot(deps) {
+    console.log('The rovpilot plugin.');
+    var self = this;
+    self.SAMPLE_PERIOD = 1000 / deps.config.sample_freq;
+
+    self.rov = deps.rov;
+    self.powerLevel = 2;
+    self.sendToROVEnabled = true;
+    self.sendUpdateEnabled = true;
+    self.power = 0.25;
+    self.priorControls = {};
+    self.vtrim = 0;
+    self.ttrim = 0;
+
+    self.positions = {
```

Comment:
Need to doublecheck where the concept of the desired position state is kept. This was in the browser javascript before so that we could meter how often the control updates were sent to the beaglebone.  Without that throttle we were spamming the node.js process with updates when using a gamepad and spiking CPU usage.  Cocepts like samplefreq are part of this abstraction that should stay browser side.